libsolidity/ast/ASTJsonImporter.cpp

```diff
@@ -348,7 +353,17 @@ ASTPointer<ContractDefinition> ASTJsonImporter::createContractDefinition(Json co
 		baseContracts,
 		subNodes,
 		contractKind(_node),
-		memberAsBool(_node, "abstract")
+		memberAsBool(_node, "abstract"),
+		nullOrCast<StorageLayoutSpecifier>(member(_node, "storageLayoutSpecifier"))
+	);
+}
+
+ASTPointer<StorageLayoutSpecifier> ASTJsonImporter::createStorageLayoutSpecifier(Json const& _node)
+{
+	astAssert(_node.contains("expression"));
+	return createASTNode<StorageLayoutSpecifier>(
+		_node,
+		std::dynamic_pointer_cast<Expression>(convertJsonToASTNode(_node["expression"]))
```

Comment:
`convertJsonToASTNode()` is already specialized to do the conversion. You just need to choose the right specialization. Doesn't this work?
```suggestion
		convertJsonToASTNode<Expression>(_node["expression"])
```